item-review-viewer/src/main/java/org/smarterbalanced/itemreviewviewer/web/services/GitLabService.java

```diff
@@ -563,32 +597,76 @@ public boolean isItemAccomExists(String itemDirName, String ext) {
+                } else {
+                    String bankKey = GitLabUtils.namespaceToBankId(namespace);
+                    tutorialId = GitLabUtils.makeDirId(bankKey, Long.toString(tutorial._id));
+                }
 
-                String tutorialId = GitLabUtils.makeDirId(Long.toString(tutorial._bankKey), Long.toString(tutorial._id));
                 if (!isItemExistsLocally(tutorialId) && downloadItem(namespace, tutorialId)) {
                     unzip(namespace, tutorialId);
                 }
 
-                if (resources != null) {
-                    for (ITSResource resource : resources) {
-                        String resourceId = GitLabUtils.makeDirId(Long.toString(resource._bankKey), Long.toString(resource._id));
-                        if (!isItemExistsLocally(resourceId) && downloadItem(namespace, resourceId)) {
-                            unzip(namespace, resourceId);
-                        }
+            }
+
+            if (resources != null) {
+                for (ITSResource resource : resources) {
+                    String resourceId;
+
+                    if(hasBankId){
+                        resourceId = GitLabUtils.makeDirId(Long.toString(resource._bankKey), Long.toString(resource._id));
+                    } else {
+                        String bankKey = GitLabUtils.namespaceToBankId(namespace);
+                        resourceId = GitLabUtils.makeDirId(bankKey, Long.toString(resource._id));
+                    }
+
+                    if (!isItemExistsLocally(resourceId) && downloadItem(namespace, resourceId)) {
+                        unzip(namespace, resourceId);
                     }
                 }
             }
+
+            if(stimulusKey > 0.0){
+                if(hasBankId) {
+                    _downloadStim(namespace, Long.toString(doc.getBankKey()), Long.toString(stimulusKey));
+                } else {
+                    _downloadStim(namespace, null, Long.toString(stimulusKey));
+                }
+            }
         } catch (Exception e) {
             throw new GitLabException(e);
         }
 
     }
 
+    //Downloads the stimulus for a given item.
+    private void _downloadStim(String namespace, String bankKey, String id){
+        String itemDirName = null;
+        if(bankKey == null && namespace != null){
+
+            bankKey = GitLabUtils.namespaceToBankId(namespace);
+        }
+        itemDirName = GitLabUtils.makeStimId(bankKey, id);
+        _logger.debug("Starting download of stim: " + id);
+
+        if(!isItemExistsLocally(itemDirName) && downloadItem(namespace, itemDirName)){
+            try {
+                unzip(namespace, itemDirName);
+                _logger.debug("Stim " + itemDirName + "successfully unzipped");
+            } catch (IOException e) {
+                _logger.error("Error processing stim: " + itemDirName);
```

Comment:
I think we should throw an error here to be caught out side of this function which is _downloadAssociatedItems. Otherwise, _downloadAssociatedItems cannot catch unzipping exception